### PR TITLE
Adding tkr-status-controller to tkr-service package

### PR DESCRIPTION
**Pull request by saji-pivotal**
### What this PR does / why we need it
Adds tkr-status controller to the tkr-service package

### Which issue(s) this PR fixes

Fixes: Related to https://github.com/vmware-tanzu/tanzu-framework/issues/1925 

### Describe testing done for PR
Manual testing
1. Applied the tkr-status-controller deployment to an existing management-cluster
2. Verified that the tkr-status-controller deployment was running as expected
3. Tested locally by deploying the tkr-service package and all the components were successfully deployed

### Release note

```release-note
package-based-lcm: Adding tkr-status-controller to tkr-service package
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by prkalle — APPROVED** [review]
LGTM. 
I would wait for @imikushin to look at it. Thanks

File: packages/management/tkr-service/bundle/config/upstream/tkr-status-controller-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    matchLabels:
+      app: tkr-status-controller
+  template:
+    metadata:
+      labels:
+        app: tkr-status-controller
+    spec:
+      containers:
+        - image: tkr-status-controller:latest
+          imagePullPolicy: Always
```
Comment (imikushin) [inline packages/management/tkr-service/bundle/config/upstream/tkr-status-controller-deployment.yaml:25]: question: 
Are we always pulling the image? Is this overridden by an overlay or something when building the package?

**Review by imikushin — APPROVED** [review]
Generally, LGTM.

With one question 🙂

File: packages/management/tkr-service/bundle/config/upstream/tkr-status-controller-deployment.yaml (hunk 1 above)
Comment (prkalle) [reply packages/management/tkr-service/bundle/config/upstream/tkr-status-controller-deployment.yaml:25]: Yes, we are not overriding the overlay.

Comment (prkalle) [reply packages/management/tkr-service/bundle/config/upstream/tkr-status-controller-deployment.yaml]: fixed: chnaged it to `IfNotPresent`

**Pull request merged**